homeassistant/components/vacuum/__init__.py

```diff
@@ -0,0 +1,287 @@
+        return []
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes of the vacuum cleaner."""
+        data = {}
+
+        if self.status is not None:
+            data['vacuum_status'] = self.status
+
+        if self.battery_level is not None:
+            data['vacuum_battery'] = self.battery_level
+            data['vacuum_battery_icon'] = self.battery_icon
+
+        if self.fanspeed is not None:
+            data['vacuum_fanspeed'] = self.fanspeed
+            data['vacuum_fanspeed_list'] = self.fanspeed_list
+
+        return data
+
+    def turn_on(self, **kwargs):
+        """Turn the vacuum on and start cleaning."""
+        raise NotImplementedError()
+
+    def async_turn_on(self, **kwargs):
+        """Turn the vacuum on and start cleaning.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        return self.hass.async_add_job(partial(self.turn_on, **kwargs))
+
+    def turn_off(self, **kwargs):
+        """Turn the vacuum off stopping the cleaning and returning home."""
+        raise NotImplementedError()
+
+    def async_turn_off(self, **kwargs):
+        """Turn the vacuum off stopping the cleaning and returning home.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        return self.hass.async_add_job(partial(self.turn_off, **kwargs))
+
+    def return_to_base(self, **kwargs):
+        """Set the vacuum cleaner to return to the dock."""
+        raise NotImplementedError()
+
+    def async_return_to_base(self, **kwargs):
+        """Set the vacuum cleaner to return to the dock.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        return self.hass.async_add_job(partial(self.return_to_base, **kwargs))
+
+    def stop(self, **kwargs):
+        """Stop the vacuum cleaner."""
+        raise NotImplementedError()
+
+    def async_stop(self, **kwargs):
+        """Stop the vacuum cleaner.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        return self.hass.async_add_job(partial(self.stop, **kwargs))
+
+    def locate(self, **kwargs):
+        """Locate the vacuum cleaner."""
+        raise NotImplementedError()
+
+    def async_locate(self, **kwargs):
+        """Locate the vacuum cleaner.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        return self.hass.async_add_job(partial(self.locate, **kwargs))
+
+    def set_fanspeed(self, **kwargs):
+        """Set the fanspeed."""
+        raise NotImplementedError()
+
+    def async_set_fanspeed(self, **kwargs):
+        """Set the fanspeed.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        return self.hass.async_add_job(partial(self.set_fanspeed, **kwargs))
+
+    def cleaning_play_pause(self, **kwargs):
+        """Pause the cleaning task or replay it."""
```

Comment:
I think I'll go with `start_pause`, stripping also the 'cleaning_' prefix (that was there from copy-pasting the media player...)